mlflow/sagemaker/__init__.py

```diff
@@ -1591,15 +1616,26 @@ def _update_sagemaker_endpoint(
     )
     _logger.info("Created new model with arn: %s", new_model_response["ModelArn"])
 
+    if not variant_name: 
+        variant_name = model_name
+
     if mode == DEPLOYMENT_MODE_ADD:
         new_model_weight = 0
         production_variants = deployed_production_variants
+        variant_name_list = [variant["VariantName"] for variant in production_variants]
+        if variant_name and variant_name in variant_name_list:
+            raise MlflowException(
+            message=(
+                "Variant {variant_name} already exists!".format(variant_name=variant_name)
+            ),
+            error_code=INVALID_PARAMETER_VALUE,
```

Comment:
Does SageMaker enforce variant name uniqueness? If not, does the client strictly need to perform this validation?

Reply:
I will look into this.

Reply:
Thanks @nfarley-soaren ! Looking forward to merging once you finish your investigation.

Reply:
Thank you @dbczumar.  Looks like Sagemaker will error out if the variant name already exists so I'll remove this part also.